tracer/src/Datadog.Trace/AppSec/Transports/Http/HttpTransport.Core.cs

```diff
@@ -40,6 +51,48 @@ public IHeadersCollection GetResponseHeaders()
         {
             return new HeadersCollectionAdapter(_context.Response.Headers);
         }
+
+        public void WriteBlockedResponse(string templateJson, string templateHtml)
+        {
+            _context.Items["block"] = true;
+            var httpResponse = _context.Response;
+            httpResponse.Clear();
+            httpResponse.Headers.Clear();
+            httpResponse.StatusCode = 403;
+            var syncIOFeature = _context.Features.Get<IHttpBodyControlFeature>();
+            if (syncIOFeature != null)
+            {
+                // allow synchronous operatoin for net core >=3.1 otherwise invalidoperation exception
+                syncIOFeature.AllowSynchronousIO = true;
+            }
+
+            var template = templateHtml;
```

Comment:
I think json would be a better default. Also, java has quite complex heuristic to decide html vs json, we might want to copy it as some point: https://github.com/DataDog/dd-trace-java/pull/3701/files#diff-1687eec313f4854c629d3e6eefeec1b3fe96531acfa074bce9333701e0bc3560R64-R97

Neither point is blocking, we could work on that after the merge.

Reply:
changed the default to json in f6f98e9fa0f87f56b9bf75cc593dc6bb6d63e5ff
will add a backlog to improve the code doing 'content negotiation' later